pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -265,15 +266,53 @@ func validatePipelineResults(results []PipelineResult) (errs *apis.FieldError) {
 		expressions = filter(expressions, looksLikeResultRef)
 		resultRefs := NewResultRefs(expressions)
 		if len(expressions) != len(resultRefs) {
-			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
+			return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
 				"value").ViaFieldIndex("results", idx))
 		}
 
+		if !TaskContainsResult(result.Value, taskMap) {
+			return errs.Also(apis.ErrInvalidValue("invalid propagation of finally task results to pipeline results",
```

Comment:
This error message is a bit confusing because we are planning to allow finally task results to be propagated to pipeline results. The actual problem seems to be that the "tasks.results.name" syntax is being used to refer to a finally task, so the error message should reflect that.

Reply:
does `referencing a nonexistent task` work better?

Reply:
yes that works!

Reply:
done.